### PR TITLE
evaluate2.sh: Add Time Limit for Runs

### PR DESCRIPTION
**Pull request by hundredwatt**
Followup to #290

This is the approach recommended by hyperfine's maintainer: https://github.com/sharkdp/hyperfine/issues/576

### CODE REVIEW

File: evaluate2.sh [hunk 1]
```diff
@@ -149,9 +150,13 @@ for fork in "$@"; do
 
     # Linux platform
     # prepend this with numactl --physcpubind=0-7 for running it only with 8 cores
-    numactl --physcpubind=0-7 hyperfine $HYPERFINE_OPTS "./calculate_average_$fork.sh 2>&1"
-  else
-    hyperfine $HYPERFINE_OPTS "./calculate_average_$fork.sh 2>&1"
+    numactl --physcpubind=0-7 hyperfine $HYPERFINE_OPTS "timeout -v $RUN_TIME_LIMIT ./calculate_average_$fork.sh 2>&1"
+  else # MacOS
```
Comment (gunnarmorling) [inline evaluate2.sh:154]: Ah, very nice. So `evaluate2.sh` runs on MacOS too?

Comment (hundredwatt) [reply evaluate2.sh:154]: Yes, I've been testing on both Fedora Linux and my MacBook M2 Air

**Comment by hundredwatt** [comment]
Since this has not been merged yet, I added a check that `bc` is installed (it was missing on a test VM I used)

File: evaluate2.sh (hunk 1 above)
Comment (AlexanderYastrebov) [reply evaluate2.sh:155]: Do we need to check whether `gtimeout` command is available?

**Comment by gunnarmorling** [comment]
> Since this has not been merged yet, I added a check that bc is installed (it was missing on a test VM I used)

Nice; just ran into that too.

File: evaluate2.sh (hunk 1 above)
Comment (gunnarmorling) [reply evaluate2.sh:155]: I have `timeout` rather than `gtimeout` on MacOS. Should we just use that?

Comment (gunnarmorling) [reply evaluate2.sh:155]: LOL, nevermind, wrong console ;)

It is checked already, so LGTM. Will merge. Will only add log message for this case.

File: evaluate2.sh [hunk 2]
```diff
@@ -149,9 +151,13 @@ for fork in "$@"; do
 
     # Linux platform
     # prepend this with numactl --physcpubind=0-7 for running it only with 8 cores
-    numactl --physcpubind=0-7 hyperfine $HYPERFINE_OPTS "./calculate_average_$fork.sh 2>&1"
-  else
-    hyperfine $HYPERFINE_OPTS "./calculate_average_$fork.sh 2>&1"
+    numactl --physcpubind=0-7 hyperfine $HYPERFINE_OPTS "timeout -v $RUN_TIME_LIMIT ./calculate_average_$fork.sh 2>&1"
```
Comment (AlexanderYastrebov) [inline evaluate2.sh:154]: I get 
```
timeout: invalid option -- 'v'
Try 'timeout --help' for more information.
```
It works without -v

Comment (AlexanderYastrebov) [reply evaluate2.sh:154]: I guess I have outdated version, I see -v here https://man7.org/linux/man-pages/man1/timeout.1.html

**Pull request merged**